src/server/storage/storage.js

```diff
@@ -357,35 +361,53 @@ Storage.prototype.loadPaths = function (data, callback) {
 
     this.mongo.openProject(data.projectId)
         .then(function (dbProject) {
-            var loadedObjects = {};
-
-            //self.logger.error('paths:', data.paths);
+            var loadedObjects = {},
+                throttleDeferred = Q.defer(),
```

Comment:
I see. I thought you were trying to throttle against too many paralel calls towards the db, but it okay then.